qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/codeactions/QuteCodeActionForUnknownProperty.java

```diff
@@ -153,4 +161,17 @@ private static void doCodeActionToCreateProperty(Part propertyPart, ResolvedJava
 
 	}
 
+	private void doCodeActionsForSimilarValues(PropertyPart part, Template template, Diagnostic diagnostic,
+			String resolvedType, List<CodeAction> codeActions) throws BadLocationException {
+		Collection<String> availableValues = collectAvailableValuesForPropertyPart(part, template, resolvedType);
+		doCodeActionsForSimilarValues(part, availableValues, template, diagnostic, codeActions);
+	}
+
+	private Collection<String> collectAvailableValuesForPropertyPart(PropertyPart node, Template template,
+			String resolvedType) {
+		String projectUri = template.getProjectUri();
+		return javaCache.resolveJavaType(resolvedType, projectUri).getNow(null).getFields().stream()

```

Comment:
You dont need to use javaCache here. You can use directly baseResolvedType

Reply:
I suggest that you see property completion at https://github.com/redhat-developer/quarkus-ls/blob/03b06df7be5f591dd597b18da9c7e7025927fa1a/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/completions/QuteCompletionsForExpression.java#L276 and  you wiĺl see that it is not pnly fields of resolved type but too fields from extended type and method which have no parameters etc